datasource.go

```diff
@@ -356,11 +359,13 @@ func (o *OCIDatasource) searchHelper(ctx context.Context, region, compartment st
 			return nil, errors.Wrap(err, "list metrircs failed")
 		}
 		items = append(items, res.Items...)
-		if res.OpcNextPage == nil {
+		// Only 0 - n-1  pages are to be fetched, as indexing starts from 0 (for page number
+		if res.OpcNextPage == nil  && pageNumber >= MAX_PAGES_TO_FETCH {
```

Comment:
I think it should be || instead of &&?

Reply:
You are right. 
Sorry, the last commit didn't get pushed. Force pushing it now